lib/spack/spack/test/package_sanity.py

```diff
@@ -172,15 +172,17 @@ def test_prs_update_old_api():
     ]
     failing = []
     for file in changed_package_files:
-        name = os.path.basename(os.path.dirname(file))
-        pkg = spack.repo.get(name)
+        filename = os.path.join(spack.paths.spack_root, file)
+        with open(filename, 'r') as f:
+            text = f.read()
 
-        # Check for old APIs
-        failed = (hasattr(pkg, 'setup_environment') or
-                  hasattr(pkg, 'setup_dependent_environment'))
+        failed = ('\n    def setup_environment' in text or
```

Comment:
(question) I feel like this approach allows testing both mock and non-mock packages, which I'm having trouble synchronizing with https://github.com/spack/spack/pull/13249#discussion_r336248474:

> I don't want to assume all builtin.mock packages are using the new API, as they could be testing the backwards compatibility to the old API.

Was there a complication with augmenting the `try`/`except` to check if the package was in the mock repo?

Reply:
The `try/except` version didn't check both, and I thought you had pushed back against my argument for not checking both. This was my workaround.